impeller/entity/contents/tiled_texture_contents.cc

```diff
@@ -228,8 +228,12 @@ std::optional<Snapshot> TiledTextureContents::RenderToSnapshot(
     bool msaa_enabled,
     int32_t mip_count,
     const std::string& label) const {
+  std::optional<Rect> geometry_coverage = GetGeometry()->GetCoverage({});
   if (GetInverseEffectTransform().IsIdentity() &&
-      GetGeometry()->IsAxisAlignedRect()) {
+      GetGeometry()->IsAxisAlignedRect() &&
+      (!geometry_coverage.has_value() ||
+       Rect::MakeSize(texture_->GetSize())
+           .Contains(geometry_coverage.value()))) {
```

Comment:
I don't fully understand this, could you explain it a bit more? I know for the regular texture contents we have some src/dst checking.

Since tiled texture contents is an arbitrary geometry, we have the axis aligned rect check first. What is the additional check doing?

Reply:
When the geometry is not contained in the texture that means we have to render texcoords outside of 0..1, so we can't take the fast pass of avoiding rendering.  We need the clamping, decal, etc.

If you check out the issue this is where we could hack this conditional to give us the correct result by sending in `Scale(1.001, 1.001, 1)` instead of `Identity()` for the effect transform because the fast pass conditional was incorrectly ignoring the geometry and the full render was correct.

Reply:
But isn't that expressed in the coverage and the scaling below?

Reply:
Demonstrably, no.  If you run the golden test without this patch it will render with the incorrect texture coords, instead stretching them to fill the whole geometry.  That flow through the code lacks the math the express the effect transform correctly.

Reply:
I think since the effect transform was identity it was assumed it could be ignored, but it's not the case when the geometry is expanded.

Reply:
Oh I agree its not working! But, this code should be able to render a stretched rectangle right? It seems like there is a bug in the math there that we could fix?

Reply:
I played around with it again and I don't think so.  The `Snapshot` flow lacks the expressivity to represent rendering texcoords outside of 0..1.  It's not a bug, but a limit of the abstraction.  That's why it only works when rendering a subset inside of the texture.

If you set the scale to 1.0 you can see it renders at the correct size without the halo we need and there is no lever to pull to make it appear.

We can file an issue to redesign `Snapshot` it will probably require having a matrix for the uvs and the vertices?

Reply:
Ack. The snapshot has a transform, but it assumes a srcRect. We could add support for that, but it isn't necessary here.